src/lsp-server/src/backend.rs

```diff
@@ -177,6 +244,39 @@ impl LanguageServer for Backend {
         // Ok(Some(lenses))
         Ok(None)
     }
+
+    /// Formats the code using [mz_sql_pretty].
+    ///
+    /// Implements the [`textDocument/formatting`](https://microsoft.github.io/language-server-protocol/specification#textDocument_formatting) language feature.
+    async fn formatting(&self, params: DocumentFormattingParams) -> Result<Option<Vec<TextEdit>>> {
+        self.client
+            .log_message(MessageType::INFO, format!("Formatting: {:?}", params))
+            .await;
+
+        let locked_map = self.parse_results.lock().await;
+        let width = self.formatting_width.lock().await;
+
+        if let Some(parse_result) = locked_map.get(&params.text_document.uri) {
```

Comment:
```suggestion
        let Some(parse_result) = locked_map.get(&params.text_document.uri) else {
            return Ok(None);
        }
```

Swap this logic so it early returns if not present. (Suggestions here will need fmt'ing.)